### ISSUE TITLE
Bard: "Your default credentials were not found."

### ISSUE THREAD

**Issue by cameronj86** [kharvd/gpt-cli#32]
Great to see Bard capabilities in the app! I'm using a similar CLI but they don't have the ability to set a consistent 'role' like this app. I've been working to get it running this afternoon and have run into stumbling locks:

1. Unlike Open_AI/AnthropicAI, the default yaml file is missing an optional line for Bard
2. The system doesn't appear to recognize when I manually add  `google_api_key:  <insert key here>` to the yaml file but...
3. It does appear to recognize `api_key: <insert key here>` instead. However, I don't get very far once within the package:

> python3 gpt-cli/gpt.py --model chat-bison-001
```
Hi! I'm here to help. Type q or Ctrl-D to exit, c or Ctrl-C to clear the        
conversation, r or Ctrl-R to re-generate the last response. To enter multi-line 
mode, enter a backslash \ followed by a new line. Exit the multi-line mode by   
pressing ESC and then Enter (Meta+Enter).         
```                              
> hello
```
An uncaught exception occurred. Please report this issue on GitHub.
Traceback (most recent call last):
  File "/home/cameron/gpt-cli/gpt.py", line 236, in <module>
    main()
  File "/home/cameron/gpt-cli/gpt.py", line 184, in main
    run_interactive(args, assistant)
  File "/home/cameron/gpt-cli/gpt.py", line 232, in run_interactive
    session.loop(input_provider)
  File "/home/cameron/gpt-cli/gptcli/session.py", line 168, in loop
    while self.process_input(*input_provider.get_user_input()):
  File "/home/cameron/gpt-cli/gptcli/session.py", line 160, in process_input
    response_saved = self._respond(args)
  File "/home/cameron/gpt-cli/gptcli/session.py", line 101, in _respond
    for response in completion_iter:
  File "/home/cameron/gpt-cli/gptcli/google.py", line 42, in complete
    response = genai.chat(**kwargs)
  File "/home/cameron/.local/lib/python3.9/site-packages/google/generativeai/discuss.py", line 342, in chat
    return _generate_response(client=client, request=request)
  File "/home/cameron/.local/lib/python3.9/site-packages/google/generativeai/discuss.py", line 478, in _generate_response
    client = get_default_discuss_client()
  File "/home/cameron/.local/lib/python3.9/site-packages/google/generativeai/client.py", line 122, in get_default_discuss_client
    default_discuss_client = glm.DiscussServiceClient(**default_client_config)
  File "/home/cameron/.local/lib/python3.9/site-packages/google/ai/generativelanguage_v1beta2/services/discuss_service/client.py", line 430, in __init__
    self._transport = Transport(
  File "/home/cameron/.local/lib/python3.9/site-packages/google/ai/generativelanguage_v1beta2/services/discuss_service/transports/grpc.py", line 151, in __init__
    super().__init__(
  File "/home/cameron/.local/lib/python3.9/site-packages/google/ai/generativelanguage_v1beta2/services/discuss_service/transports/base.py", line 97, in __init__
    credentials, _ = google.auth.default(
  File "/home/cameron/.local/lib/python3.9/site-packages/google/auth/_default.py", line 648, in default
    raise exceptions.DefaultCredentialsError(_CLOUD_SDK_MISSING_CREDENTIALS)
google.auth.exceptions.DefaultCredentialsError: Your default credentials were not found. To set up Application Default Credentials, see https://cloud.google.com/docs/authentication/external/set-up-adc for more information.
```

I followed the instructions [via the error msg's link](https://cloud.google.com/docs/authentication/provide-credentials-adc#how-to) & download the gcloud cli package. Once this step is completely, I also manually set the api-key within the gcloud-cli as well but the issue persists.

**Comment by kharvd** [comment]
Could you please paste your full config file and its location? (Just make sure to exclude the actual API keys) It looks like the api key is not being set correctly for some reason

**Comment by cameronj86** [comment]
[Here ya go](https://i.habedieeh.re/O7iovn8.png) and the path is ~/.config/gpt-cli/gpt.yml

**Comment by cameronj86** [comment]
Whoops, here's a [working link](https://i.imgur.com/gBPvjFt.png)

**Comment by kharvd** [comment]
Everything below the API keys looks like a stub from the README - you don't have to have that (and it might actually break things). Could you try removing all of that and just leave the keys?

**Comment by cameronj86** [comment]
Yeah I just copy/pasted the ReadMe 😅. In retrospect, I should've realized that wasn't the purpose once I received error messages for most rows that listed the options 

Just now, I did try commenting out everything save for the api_key and still no dice.

I decided to take a step back and just export the key to .bashrc instead, and received the following error:
>google.api_core.exceptions.PermissionDenied: 403 Generative Language API has not been used in project <personal api key project id> before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/generativelanguage.googleapis.com/overview?project=<personal api key project id> then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry. [links {
  description: "Google developers console API activation"
  url: "https://console.developers.google.com/apis/api/generativelanguage.googleapis.com/overview?project=<personal api key project id>"

Did some googling and it looks like someone stumbled into the same issue today:
https://stackoverflow.com/questions/76244554/gcp-generativelanguage-googleapis-com-not-able-to-enable

**TL;DR:** From the sound of things, I need to be a part of a private beta to use the google api w/ Bard

**Comment by adiso06** [comment]
any idea if this works now? I'm still getting the same issue, but i think this waitlist is no longer an issue?
